src/pathpyG/core/MultiOrderModel.py

```diff
@@ -278,3 +282,280 @@ def from_DAGs(
+        ), "Error: max_order cannot be larger than maximum order of multi-order network"
+
+        dof = self.layers[1].data.num_nodes - 1  # Degrees of freedom for zeroth order
+
+        if assumption == "paths":
+            # COMPUTING CONTRIBUTION FROM NUM PATHS AND NONERO OUTDEGREES SEPARATELY
+            # TODO: CAN IT BE DONE TOGETHER?
+
+            edge_index = self.layers[1].data.edge_index
+            # Adding dof from Number of paths of length k
+            for k in range(1, max_order + 1):
+                if k > 1:
+                    num_nodes = 0 if edge_index.numel() == 0 else edge_index.max().item() + 1
+                    edge_index = self.lift_order_edge_index(edge_index, num_nodes)
+                #counting number of len k paths
+                num_len_k_paths = edge_index.shape[1]#edge_index.max().item() +1  # Number of paths of length k
+                dof += num_len_k_paths
+
+            # removing dof from total probability of nonzero degree nodes
+            for k in range(1, max_order + 1):
+                if k == 1:
+                    edge_index_adj = self.layers[1].data.edge_index
+                    edge_index = edge_index_adj
+                else:
+                    edge_index, _ = edge_index @ edge_index_adj
+                num_nonzero_outdegrees = torch.unique(edge_index[0]).size(0)
+                dof -= num_nonzero_outdegrees
+
+        elif assumption == "ngrams":
+            for order in range(1, max_order + 1):
+                dof += (self.layers[1].data.num_nodes ** order) * (self.layers[1].data.num_nodes - 1)
+        else:
+            raise ValueError(
+                f"Unknown assumption {assumption} in input. The only accepted values are 'path' and 'ngram'"
+            )
+
+        return int(dof)
+
+    def get_zeroth_order_log_likelihood(self, dag_graph: DataBatch) -> float:
+        """
+        Compute the zeroth order log likelihood.
+
+        Args:
+            dag_graph (DataBatch): Input DAG graph data.
+
+        Returns:
+            float: Zeroth order log likelihood.
+        """
+        # Get frequencies
+        # TODO: put this tensor directly in dag_graph (intead of edge_weight) and remove the following line
+        # getting the index of the last edge of each path (to be used to extract weights)
+        last_edge_ixs = dag_graph.ptr[1:] - torch.arange(2,dag_graph.ptr.shape[0]+1)
+        frequencies = dag_graph.edge_weight[last_edge_ixs]
+
+        # Get ixs starting nodes
+        # Q: Is dag_graph.ptr[:-1] enough to get the start_ixs?
+        mask = torch.ones(dag_graph.num_nodes, dtype=bool)
+        mask[dag_graph.edge_index[1]] = False
+        start_ixs = dag_graph.node_sequence.squeeze()[mask]
+
+        # Compute node emission probabilities
+        # TODOL modify once we have zeroth order in mon
+        _, counts = torch.unique(dag_graph.node_sequence, return_counts=True)
+        node_emission_probabilities = counts / counts.sum()
```

Comment:
We discussed this in our meeting and agreed to ignore it intentionally. I still think we should at least add it as comment so that we won't forget about it.